### ISSUE TITLE
Bug Report: Synchronize module asks password

### ISSUE THREAD

**Issue by BR0kEN-** [ansible/ansible-modules-core#2502]
This is duplicate of [#5934](https://github.com/ansible/ansible/issues/5934)!
##### Issue Type:

Bug Report
##### Ansible Version:

1.9.4
##### Ansible Configuration:

Original.
##### Environment:

OS X Yosemite 10.10.5 (14F1021)
##### Summary:

inventory

``` ini
[vagrant]
192.168.56.132 ansible_ssh_user=vagrant ansible_ssh_private_key_file=.vagrant/machines/default/virtualbox/private_key
```

provision.yml

``` yml
- hosts: all
  gather_facts: yes
  sudo: yes

tasks:
  - synchronize:
      src: ./
      dest: "/var/lib/jenkins"
      archive: no
      recursive: yes
    sudo: yes
```
##### Steps To Reproduce:

``` shell
ansible-playbook -vvvv provision.yml -i inventory --limit=vagrant
```
##### Expected Results:

`synchronize` should not ask for a password, as other tasks do.
##### Actual Results:

```
<127.0.0.1> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1447416965.06-222404616716035 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1447416965.06-222404616716035 && echo $HOME/.ansible/tmp/ansible-tmp-1447416965.06-222404616716035']
<127.0.0.1> PUT /var/folders/72/r_f79s2974q5b59cq0f7c1wh0000gn/T/tmpTRff6u TO /Users/BR0kEN/.ansible/tmp/ansible-tmp-1447416965.06-222404616716035/synchronize
<127.0.0.1> EXEC ['/bin/sh', '-c', u'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /Users/BR0kEN/.ansible/tmp/ansible-tmp-1447416965.06-222404616716035/synchronize; rm -rf /Users/BR0kEN/.ansible/tmp/ansible-tmp-1447416965.06-222404616716035/ >/dev/null 2>&1']
vagrant@192.168.56.132's password:
```
##### Debugging steps:

I've go to `ansible/modules/core/files/synchronize.py` and put `print(cmdstr)` before `(rc, out, err) = module.run_command(cmd)`. Module generated the next command:

``` shell
rsync --delay-updates -F --compress --recursive --rsh 'ssh -i .vagrant/machines/default/virtualbox/private_key -S none -o StrictHostKeyChecking=no' --rsync-path="sudo rsync" --out-format='<<CHANGED>>%i %n%L' "/Users/BR0kEN/cibox/aae/provisioning/roles/cibox-jenkins/files/" "vagrant@192.168.56.132:/var/lib/jenkins"
```

**And even more:**

Open the `ansible/runner/__init__.py` and put `print(cmd)` before `res = self._low_level_exec_command(conn, cmd, tmp, become=self.become, sudoable=sudoable, in_data=in_data)`. Module generated something like:

``` shell
LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /Users/BR0kEN/.ansible/tmp/ansible-tmp-1447418694.04-11945943198607/synchronize; rm -rf /Users/BR0kEN/.ansible/tmp/ansible-tmp-1447418694.04-11945943198607/ >/dev/null 2>&1
```

When I execute one of these raw commands, then everything fine. 

Future debugging said that problem exists with VirtualBox (Vagrant). When installing on a remote - everything is fine, but not on mounted (NFS) file system.

**Comment by BR0kEN-** [comment]
I guess that new local connection should not be opened for that task and command must be executed in current shell (because `rsync` has ssh credentials to sync to remote (Vagrant)).

**Comment by robynbergeron** [comment]
[module: files/synchronize.py]

**Comment by robynbergeron** [comment]
@BR0kEN- Thanks for submitting this bug report.

@tima -- thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module.

**Comment by tima** [comment]
This is a known issue that has culminated in https://github.com/ansible/ansible/issues/13099 under the core project.

**Comment by goozbach** [comment]
I've got a test case which 100% fails.

https://github.com/goozbach/ansible-broken-sync-vagrant/tree/master

**Comment by tima** [comment]
@goozbach We know there are cases where it 100% fails that's why it's a known issue. It's fixing it reliably and securely in all modules that need to do an extra level of authentication is a different matter.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by tima** [comment]
I cannot reproduce this error with 2.1.1.0 so I'm going to assume it was fixed. Can you please verify @goozbach?

...

> ==> default: Machine booted and ready!
> ==> default: Checking for guest additions in VM...
>     default: No guest additions were detected on the base box for this VM! Guest
>     default: additions are required for forwarded ports, shared folders, host only
>     default: networking, and more. If SSH fails on this machine, please install
>     default: the guest additions and repackage the box to continue.
>     default: 
>     default: This is not an error message; everything may continue to work properly,
>     default: in which case you may ignore this message.
> ==> default: Rsyncing folder: /Users/tima/tmp/ansible-broken-sync-vagrant/ => /home/vagrant/sync
> ==> default: Running provisioner: ansible...
>     default: Running ansible-playbook...
> 
> PLAY [test broke sync] *********************************************************
> 
> TASK [setup] *******************************************************************
> ok: [default]
> 
> TASK [sync foo] ****************************************************************
> changed: [default]
> 
> PLAY RECAP *********************************************************************
> default                    : ok=2    changed=1    unreachable=0    failed=0   
> 
> (ansible-2.1.1.0) 

waiting_on_contributor

**Comment by tima** [comment]
needs_contributor

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by jctanner** [comment]
superseded by https://github.com/ansible/ansible/issues/13099
